### ISSUE TITLE
pkgs.hplipWithPlugin: url is no longer correct

### ISSUE THREAD

**Issue by eqyiel** [NixOS/nixpkgs#31262]
## Issue description

I noticed that the hash was changing every time I tried to build it.  At first I thought something nefarious was going on but I guess Nix is downloading the webpage that you see if you visit `http://hplipopensource.com/hplip-web/plugin/3.17.9-plugin.run`.

It looks like you can now get the file from sourceforge, but I can't figure out how to get a direct link to the file.

cc @ttuegel because you have worked on this package in the past

### Steps to reproduce

`nix-build '<nixpkgs>' -A pkgs.hplipWithPlugin`

## Technical details

* System: 18.03.git.02a33d2325 (Impala)
* Nix version: nix-env (Nix) 1.11.15
* Nixpkgs version: "18.03.git.56a34abf5fM"
* Sandboxing enabled: false

**Comment by vcunat** [comment]
> It looks like you can now get the file from sourceforge, but I can't figure out how to get a direct link to the file.

We have lots of `mirror://sourceforge/...` in nixpkgs, so I'd assume it would work here as well.

**Comment by bugaev** [comment]
I confirm, I couldn't install system with hplip. Proceeded without until solved.

**Comment by eqyiel** [comment]
It looks like this retrieves the right file, but the build fails because the `plugin_tmp` dir in the postInstall script doesn't exist.

```diff
   plugin = fetchurl {
-    url = "http://hplipopensource.com/hplip-web/plugin/${name}-plugin.run";
-    sha256 = "10z8vzwcwmwni7s4j9xp0fa7l4lwrhl4kp450dga3fj0cck1gxwq";
+    url = "mirror://sourceforge/hplip/${name}.run";
+    sha256 = "1hvv6m4yc7lsw6d0zing091qrr9ihagv4ir58mdif5w0aixks5yw";
   };
```

**Comment by eqyiel** [comment]
This is fixed by #31287, thanks @ttuegel
